src/hls_lpdaac_reconciliation/response/__init__.py

```diff
@@ -100,6 +100,19 @@ def group_granule_ids(
     }
 
 
+def granule_id_for_file(filename: str) -> str:
+    """Determine the granule ID for a file."""
+    import re
+
+    # Example: HLS.S30.T15XWH.2024237T194859.v2.0_stac.json
+    # Match everything through to the version number vX[.Y[...]] (e.g., v2.0),
+    # ignoring all characters following the version number.
+    if not (m := re.match(r"^(?P<granule_id>.+\.v\d(?:\.\d)*)", filename)):
```

Comment:
perf nit - might prefer to `re.compile` and re-use since this is inside of a loop

Reply:
Good catch! Will do...